packages/docusaurus/src/server/index.ts

```diff
@@ -150,6 +151,17 @@ export async function load(
       .join('\n')}\n];\n`,
   );
 
+  const rootWrappers = loadRootWrappers(plugins);
+  const genRootWrappers = generate(
+    generatedFilesDir,
+    'root-wrapper.js',
+    `export default [\n${rootWrappers
+      // import() is async so we use require() because client modules can have
+      // CSS and the order matters for loading CSS.
+      // We need to JSON.stringify so that if its on windows, backslash are escaped.
+      .map((module) => `  require(${JSON.stringify(module)}),`)
+      .join('\n')}\n];\n`,
+  );
```

Comment:
That's an interesting api.

We'll need it for CSS-in-JS too so we should try to make sure this API will be good enough, to avoid future breaking changes. 
Also worth comparing to what Gatsby does

https://github.com/facebook/docusaurus/issues/3236

Reply:
I got it from gatsby only :)